tests/UnifiedSpecTests/Operation.php

```diff
@@ -517,14 +517,14 @@ private function executeForCollection(Collection $collection)
                 assertArrayHasKey('out', $args);
                 assertInstanceOf(Javascript::class, $args['map']);
                 assertInstanceOf(Javascript::class, $args['reduce']);
-                assertIsString($args['out']);
+                assertThat($args['out'], logicalOr(new IsType('string'), new IsType('array'), new IsType('object')));
 
-                return $collection->mapReduce(
+                return iterator_to_array($collection->mapReduce(
```

Comment:
Without this, the test runner attempted to match a MapReduceResult directly and timed out (likely due to some loop when dumping the comparison failure). This is consistent with eager iteration for other cursor-returning operations.

While looking into this, I wondered by we're still handling write result classes in ExpectedResult instead of immediately after an Operation. Opened [PHPLIB-1406](https://jira.mongodb.org/browse/PHPLIB-1406) to revisit that.